src/config/env.ts

```diff
@@ -1,103 +1,105 @@
-
-function getEnvVar(varName: keyof PagoPAEnvVars, devVarName: string): string {
-  return isProduction && PAGOPA_ENV ? PAGOPA_ENV[varName] : `${DEV_API_HOST_URL}/${devVarName}`
+const parseAppMode = (): ViteConfigMode => {
+  return viteConfigMode.parse(import.meta.env.MODE)
 }
 
-const SERVICE_VERSION = import.meta.env.REACT_APP_SERVICE_VERSION
-export const API_GATEWAY_INTERFACE_URL = getEnvVar(
-  'API_GATEWAY_INTERFACE_URL',
-  'swagger/docs/interface-specification.yml'
-)
-export const BACKEND_FOR_FRONTEND_URL = getEnvVar(
-  'BACKEND_FOR_FRONTEND_URL',
-  `${SERVICE_VERSION}/backend-for-frontend`
-)
-export const AUTHORIZATION_SERVER_TOKEN_CREATION_URL = getEnvVar(
-  'AUTHORIZATION_SERVER_TOKEN_CREATION_URL',
-  'authorization-server/token.oauth2'
-)
-export const CLIENT_ASSERTION_JWT_AUDIENCE =
-  isProduction && PAGOPA_ENV ? PAGOPA_ENV.CLIENT_ASSERTION_JWT_AUDIENCE : ''
+const parseConfigs = (): InteropFEConfigs => {
+  // Take the configs from the window object
+  const configs = transformedFEConfigs.safeParse((window as unknown as ExtendedWindow).pagopa_env)
+  if (!configs.success) {
+    mapAndThrowZodConfigsError(configs.error)
+  }
 
-function getWellKnownUrls(wellKnownUrls: string | undefined) {
-  return wellKnownUrls?.split(',').filter((url) => !!url) || []
+  return configs.data!
 }
 
-export const WELL_KNOWN_URLS =
-  isProduction && PAGOPA_ENV ? getWellKnownUrls(PAGOPA_ENV.WELL_KNOWN_URLS) : ['#']
-
-export const SELFCARE_BASE_URL =
-  isProduction && PAGOPA_ENV ? PAGOPA_ENV.SELFCARE_BASE_URL : 'https://uat.selfcare.pagopa.it'
-
-export const STAGE = PAGOPA_ENV?.STAGE ?? 'DEV'
-
-export const PRODUCER_ALLOWED_ORIGINS = PAGOPA_ENV?.PRODUCER_ALLOWED_ORIGINS.split(',')
-  .map((o) => o.trim())
-  .filter(Boolean) ?? ['IPA']
+const mapAndThrowZodConfigsError = (errors: z.ZodError<InteropFEConfigs>) => {
+  const invalidEnvVars = errors.issues.flatMap((issue) => issue.path + ':' + issue.message)
+  throw new Error(`Invalid or missing env vars within FE-interop app: ${invalidEnvVars}`)
+}
 
-export const API_SIGNAL_HUB_PUSH_INTERFACE_URL =
-  isProduction && PAGOPA_ENV
-    ? PAGOPA_ENV['API_SIGNAL_HUB_PUSH_INTERFACE_URL']
-    : `https://raw.githubusercontent.com/pagopa/interop-signalhub-core/refs/heads/main/docs/openAPI/push-signals.yaml`
+export const {
+  BACKEND_FOR_FRONTEND_URL,
+  SELFCARE_LOGIN_URL: FE_LOGIN_URL,
+  SELFCARE_BASE_URL,
+  API_GATEWAY_INTERFACE_URL,
+  AUTHORIZATION_SERVER_TOKEN_CREATION_URL,
+  CLIENT_ASSERTION_JWT_AUDIENCE,
+  API_SIGNAL_HUB_PULL_INTERFACE_URL,
+  API_SIGNAL_HUB_PUSH_INTERFACE_URL,
+  INTEROP_RESOURCES_BASE_URL,
+  MIXPANEL_PROJECT_ID,
+  ONETRUST_DOMAIN_SCRIPT_ID,
+  WELL_KNOWN_URLS,
+  PUBLIC_URL,
+  STAGE,
+  PRODUCER_ALLOWED_ORIGINS,
+  TEMP_USER_BLACKLIST_URL,
+  FEATURE_FLAG_SIGNALHUB_WHITELIST,
+  M2M_JWT_AUDIENCE,
+  SIGNALHUB_WHITELIST_CONSUMER,
+  SIGNALHUB_WHITELIST_PRODUCER,
+  NODE_ENV,
+} = parseConfigs()
+
+export const APP_MODE = parseAppMode()
```

Comment:
Maybe there is no need of the function since it is parsed just one time
```suggestion
export const VITE_CONFIG_MODE = viteConfigMode.parse(import.meta.env.MODE)
```